lib/org/chromium/webidl/IDLFragmentExtractor.js

```diff
@@ -0,0 +1,101 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'IDLFragmentExtractor',
+  documentation: 'extracts IDL Fragments from HTML files',
+  requires: [
+    'foam.parsers.html.HTMLLexer'
+  ],
+
+  properties: [
+    {
+      name: 'file',
+      of: 'org.chromium.webidl.HTMLFileContent',
+      postSet: function(_, file) {
+        this.idlFragments = this.extract();
+      },
+    },
+    {
+      name: 'idlFragments',
+      // description: 'all idlFragments found in the parsed HTML file',
+    },
+  ],
+
+  methods: [
+    function extract() {
+      var self = this;
+      var lexer = self.HTMLLexer.create();
+      var OPEN = lexer.TagType.OPEN.name;
+      var CLOSE = lexer.TagType.CLOSE.name;
+      var extractAttr = function(node, attrName) {
```

Comment:
I have made minor changes to the extract code which allows for this. It will be part of the next set of changes.